src/registrar/views/domain.py

```diff
@@ -630,6 +632,67 @@ class DomainUsersView(DomainBaseView):
 
     template_name = "domain_users.html"
 
+    def get_context_data(self, **kwargs):
+        """The initial value for the form (which is a formset here)."""
+        context = super().get_context_data(**kwargs)
+
+        # Add conditionals to the context (such as "can_delete_users")
+        context = self._add_booleans_to_context(context)
+
+        # Add modal buttons to the context (such as for delete)
+        context = self._add_modal_buttons_to_context(context)
+
+        # Get the email of the current user
+        context["current_user_email"] = self.request.user.email
+
+        return context
+
+    def _add_booleans_to_context(self, context):
+        # Determine if the current user can delete managers
+        domain_pk = None
+        can_delete_users = False
+
+        if self.kwargs is not None and "pk" in self.kwargs:
+            domain_pk = self.kwargs["pk"]
+            # Prevent the end user from deleting themselves as a manager if they are the
+            # only manager that exists on a domain.
+            can_delete_users = UserDomainRole.objects.filter(domain__id=domain_pk).count() > 1
+
+        context["can_delete_users"] = can_delete_users
+        return context
+
+    def _add_modal_buttons_to_context(self, context):
+        """Adds modal buttons (and their HTML) to the context"""
+        # Create HTML for the modal button
+        modal_button = self._create_modal_button_html(
+            button_name="delete_domain_manager",
+            button_text_content="Yes, remove domain manager",
+            classes=["usa-button", "usa-button--secondary"],
+        )
+        context["modal_button"] = modal_button
+
+        # Create HTML for the modal button when deleting yourself
+        modal_button_self = self._create_modal_button_html(
+            button_name="delete_domain_manager_self",
+            button_text_content="Yes, remove myself",
+            classes=["usa-button", "usa-button--secondary"],
+        )
+        context["modal_button_self"] = modal_button_self
+
+        return context
+
+    def _create_modal_button_html(self, button_name: str, button_text_content: str, classes: List[str] | str):
```

Comment:
I think this level of deconstruction is actually making the code harder to read, not easier. I'd merge _add and _create together

Reply:
To generalize or not to generalize, that is the question